### PR TITLE
Fix distortion test plugin and map gen compilation

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Fix compile errors in `DistortionTestPlugin` and `mapgen`
- Update `DistortionTestPlugin` with TeamSpeak3 Channel Switching functionalities (copy)
- Add omni-directional support to `mapgen`

### CODE REVIEW

File: extensions/src/ACRE2Arma/signal/signal.hpp [hunk 1]
```diff
@@ -239,6 +239,7 @@ namespace acre {
 
                 float lower_sensitivity = args_.as_float();
                 float upper_sensitivity = args_.as_float();
+                bool omnidirectional = args_.as_int();
```
Comment (jonpas) [inline extensions/src/ACRE2Arma/signal/signal.hpp:242]: Need to use `bool omnidirectional = args_.as_int() == 1;`.

**Review by TheMagnetar — CHANGES_REQUESTED** [review]
I do not think that this  is necessary in c++. 1 is already "true". There is no difference.

**Comment by jonpas** [comment]
It is done the same way on L108, I think doing it explicitly might be cleaner for consistency as it is already used like that. Either way, something for another PR if we want improvement on it.

File: extensions/src/ACRE2Arma/signal/signal.hpp [hunk 2]
```diff
@@ -355,7 +356,7 @@ namespace acre {
             }
 
             void signal_map_chunk(std::vector<signal_map_result> *results, float start_y, float start_x, uint32_t y_offset, uint32_t length, uint32_t x_size, float sample_size,
-                float rx_antenna_height, glm::vec3 tx_pos_, glm::vec3 tx_dir_, antenna_p &tx_antenna_, antenna_p &rx_antenna_, float frequency_, float power_, float sinad_
+                float rx_antenna_height, glm::vec3 tx_pos_, glm::vec3 tx_dir_, antenna_p &tx_antenna_, antenna_p &rx_antenna_, float frequency_, float power_, float sinad_, bool omnidirectional_
```
Comment (jonpas) [inline extensions/src/ACRE2Arma/signal/signal.hpp:359]: Would actually be good to use that new variable on L373...

**Comment by jonpas** [comment]
Tested and works, good to go.

**Review by TheMagnetar — APPROVED** [review]

**Pull request merged**